detanalysis/photon_calibration.py

```diff
@@ -0,0 +1,2448 @@
+            plt.ylabel("Counts Per Bin")
+            plt.show()
+        
+        cut_pars = {'val_lower': peak_center - cut_width, 'val_upper': peak_center + cut_width,}
+	
+        import detanalysis as da
+        photon_cut = da.Semiautocut(self.calibration_df, cut_rq = cut_rq_name_mod,
+                                   channel_name=self.channel_name,
+                                   cut_pars=cut_pars, cut_name=cut_name)
+        _ = photon_cut.do_cut(lgcdiagnostics=lgc_diagnostics)
+        
+        if lgc_plot==True:
+            photon_cut.plot_histograms()
+            
+        self.photon_cut_dict[peak_number] = cut_name
+            
+    def load_events(self, photon_peak_number, number_events_limit=1000,
+                    pretrigger_window=None, trace_length=None,  
+                    raw_path=None, 
+                    lgc_plot_example_events=True, example_events_num=5,
+                    lgc_filter_plot=True, filter_freq=50e3):
+        """
+        Defines a region to accept events from for later pulse shape analysis,
+        and creates a cut using Semiautocuts to use to pull events from later.
+        Must be run after the peaks are fit using fit_spectrum.
+        
+        Parameters
+        ----------
+        photon_peak_number : int
+            The key for the photon_cut_dict used to get the cut name for the
+            photon peak cut. Is usually just '0' or '2'.
+            
+        number_events_limit : int, optional
+            The maximuim number of events to load. If there's less than this
+            passing all cuts, all events passing cuts will be loaded. Otherwise
+            the selection is random (per detanalysis).
+            
+        pretrigger_window : float, optional
+            If not None, overwrites the object's stored pretrigger window. In
+            units of milliseconds.
+            
+        trace_length : float, optional
+            If not None, overwrites the object's stored trace length. In units
+            of milliseconds.
+            
+        lgc_plot_example_events : bool, optional
+            If True, displays plots showing example events.
+            
+        example_event_number : int, optional
+            If lgc_plot_example_events is True, shows this number of example
+            events.
+            
+        lgc_filter_freq : bool, optional
+            If True, filters the plotted example events with a low pass filter
+            
+        filter_freq : float, optional
+            The low pass filter frequency for the displayed pulses
+        """ 
+        
+        if raw_path is not None:
+            self.traces_raw_path = raw_path
+        
+        if pretrigger_window is not None:
+            self.pretrigger_window = pretrigger_window
+        else:
+            pretrigger_window = self.pretrigger_window
+            
+        if trace_length is not None:
+            self.trace_length = trace_length
```

Comment:
You default to 10ms when you initialize the class variable, since there is no option to pass the trace length from the user in the class definition (from what I have seen). Otherwise you would write something like:
`def __init__(self, template_model, photon_energy_ev, ..., trace_length=10e-3, pretrigger_window=None):
...
self.trace_length = trace_length`
Where at this point if no trace length was given by the user it is defaulting to 10 ms, but if one was given it is stored as the self variable and overwrites the default. Same would be done for pretrigger, but you could set the default as half of the trace length via:
`self.pretrigger_window = pretrigger_window
...
if pretrigger_window is None:
pretrigger_window = trace_length/2`